services/comments.js

```diff
@@ -110,6 +111,7 @@ module.exports = class CommentsService {
       $set: {
         body,
         status,
+        metadata: dotize(metadata),
```

Comment:
This doesn't work, the resulting $set object should be:
```
$set {
  body: '',
  status: '',
  'metadata.settingA': 'valueA',
  'metadata.settingB': 'valueB',
}
```

Check out what `dotize` does. You might need to extend it's functionality.